### PR TITLE
Use the power of makefile to install the pim assets

### PR DESCRIPTION
**Pull request by arnolanglade**
Before reviewing this PR you should have a look to https://github.com/akeneo/pim-community-dev/pull/9677

**Description**

What the command pim:installer:asset does ?
* Dump require JS paths
* Dump JS form extensions
* Copy all assets from bundles to web folder
* Dump JS translations
* Dump js routes

Do I need to run these commands every time? No!
* I only need to dump require JS paths if I changed/added any `requirejs.yml` file.
* I only need to dump JS form extensions if I changed/added any `form_extensions.yml` file.
* I only need to dump copy assets if I changed/added any file located `/Resources/public/` from my bundle.
* I only need to dump JS translations if I changed/added any `jsmessages*.yml`
* It depends on the routes stored in the Symfony container, so unfortunately we need to run this command all time to be sure the routes are up to date.

We can use the power of makefile to avoid to run command when it is not necessary. A target won't be executed if the prerequisite did not change. 

Let's take a simple example: dumping require JS path. First I will create a new target to dump them.
```
web/js/require-paths.js:
	$(PHP_EXEC) bin/console pim:installer:dump-require-paths
```

But, I want to update the `web/js/require-paths.js` file only if I made some changes in any `requirejs.yml` files. If my target has every `requirejs.yml`files as prerequisite the command `pim:installer:dump-require-paths` will be only run when a `requirejs.yml` file has been changed. Thanks to the shell command `find . -name "requirejs.yml"` I am able to find them.

```
web/js/require-paths.js: $(shell find . -name "requirejs.yml")
	$(PHP_EXEC) bin/console pim:installer:dump-require-paths
```

Now my require js path are dump when it is needed thank to makefile!

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -83,13 +90,30 @@ vendor: composer.lock
...
+	$(PHP_EXEC) bin/console pim:installer:dump-require-paths
+
+web/js/extensions.json: $(FORM_EXTENSION_FILES)
+	$(PHP_EXEC) bin/console pim:installer:dump-extensions
+
+web/bundles: $(ASSET_FILES)
+	$(PHP_EXEC) bin/console assets:install --relative --symlink
+
+web/js/translation:
+	$(PHP_EXEC) bin/console oro:translation:dump 'en_US, ca_ES, da_DK, de_DE, es_ES, fi_FI, fr_FR, hr_HR, it_IT, ja_JP, nl_NL, pl_PL, pt_BR, pt_PT, ru_RU, sv_SE, tl_PH, zh_CN, sv_SE, en_NZ'
```
Comment (anaelChardan) [inline Makefile]: do we need all of those?

Comment (arnolanglade) [reply Makefile]: What do you mean? do you speak about the list of locale? yes we need to specify the list of locale, the command need them. For now, they are stored in a yaml file but I did not want to parse a yaml file in the makefile.

Comment (anaelChardan) [reply Makefile]: My question was "could we just generate the en_US / fr_FR / de_DE and es_ES" as they are the only one we use?

Comment (arnolanglade) [reply Makefile]: The PIM does not work like that for now, I did want to change the behavior in this PR.

Comment (ahocquard) [reply Makefile]: keep it simple at first, let's iterate on it later on

**Pull request merged**